poseidon2/src/matrix.rs

```diff
@@ -1,23 +1,23 @@
 use p3_field::{AbstractField, PrimeField};
+use p3_mds::MdsPermutation;
 use p3_symmetric::Permutation;
 
 extern crate alloc;
 
-// The t x t matrix M_E := circ(2M_4, M_4, ..., M_4), where M_4 is the 4 x 4 matrix
+// For the external layers we use a matrix of the form circ(2M_4, M_4, ..., M_4)
+// Where M_4 is a 4 x 4 MDS matrix. This leads to a permutation which has slightly weaker properties to MDS
+pub trait MDSLightPermutation<T: Clone, const WIDTH: usize>: Permutation<[T; WIDTH]> {}
```

Comment:
Couple nits,

- I would use `///` so the comment shows up in docs, IDE hints etc.
- There's a [convention](https://rust-lang.github.io/api-guidelines/naming.html) for camel case acronyms, like `MdsLightPermutation`